packages/back-end/src/api/environments/listEnvironments.ts

```diff
@@ -0,0 +1,41 @@
+import { ListEnvironmentsResponse } from "../../../types/openapi";
+import { createApiRequestHandler } from "../../util/handler";
+import { findOrganizationById } from "../../models/OrganizationModel";
+import { listEnvironmentsValidator } from "../../validators/openapi";
+
+export const listEnvironments = createApiRequestHandler(
+  listEnvironmentsValidator
+)(
+  async (req): Promise<ListEnvironmentsResponse> => {
+    const id = req.params.id;
+
+    const org = await findOrganizationById(id);
+    if (!org) {
+      throw Error("Organization not found");
+    }
```

Comment:
Do not let the user specify the org.  The org should always come from the request context.  With this code, any user could get the list of environments from any other org just by knowing the org's id.

Reply:
That's a mistake thanks for spotting it.